addons/info/src/components/Story.js

```diff
@@ -382,7 +383,7 @@ Story.propTypes = {
   showInline: PropTypes.bool,
   showHeader: PropTypes.bool,
   showSource: PropTypes.bool,
-  styles: PropTypes.func.isRequired,
+  styles: PropTypes.object, // eslint-disable-line react/forbid-prop-types
```

Comment:
Let's rather make it `PropTypes.oneOfType(PropTypes.func, PropTypes.object)` and add a runtime check. If user supplies a function we call it, otherwise do a deep merge.

That way we won't introduce any breaking changes, all the old code will keep working

Reply:
good call, will update

Reply:
Oh, we actually do that runtime check already:
https://github.com/storybooks/storybook/blob/master/addons/info/src/index.js#L56

Let's just replace the default `s => s` function with deep merge